beacon-chain/node/options.go

```diff
@@ -1,14 +1,25 @@
 package node
 
-import "github.com/prysmaticlabs/prysm/beacon-chain/blockchain"
+import (
+	"github.com/prysmaticlabs/prysm/beacon-chain/blockchain"
+	"github.com/prysmaticlabs/prysm/beacon-chain/powchain"
+)
 
 // Option for beacon node configuration.
 type Option func(bn *BeaconNode) error
 
 // WithBlockchainFlagOptions includes functional options for the blockchain service related to CLI flags.
 func WithBlockchainFlagOptions(opts []blockchain.Option) Option {
 	return func(bn *BeaconNode) error {
-		bn.blockchainFlagOpts = opts
+		bn.serviceFlagOpts.blockchainFlagOpts = opts
+		return nil
```

Comment:
How are you feeling about these returning an error? It really is starting to feel like the utility of that is super limited. I need to go back and see why I suggested we need this, I think it had something to do with letting service opts raise errors at the point where they are used (ie a opt that returns a file reader raising an os error to be returned by `NewService`.

Reply:
yep it is useful in the p2p service I believe because you can do some validity checks in there. It's easier to keep the pattern of returning and error and returning nil rather than not returning error and having to refactor all packages because we suddenly need it. Similar to passing ctx as the first argument into functions